src/main/java/com/google/devtools/build/lib/actions/ActionCacheChecker.java

```diff
@@ -304,6 +305,12 @@ protected boolean mustExecute(
       actionCache.accountMiss(MissReason.DIFFERENT_ENVIRONMENT);
       return true;
     }
+    if (!Objects.equals(entry.getRemoteDefaultPlatformProperties(),
+        actionKeyContext.getRemoteDefaultPlatformProperties())) {
+      reportClientEnv(handler, action, usedClientEnv);
+      actionCache.accountMiss(MissReason.DIFFERENT_ENVIRONMENT);
```

Comment:
maybe introduce a new reason?

Reply:
sure, just wanted to make sure that it works this way first, and it is!

I also spend some time trying to avoid using `actionKeyContext` - no good alternative so far - requires tooo many changes (like a lot). Will try add it to the `BuildOptions` maybe in the next experiment's round.

Reply:
sure, just wanted to make sure that it works this way first, and it is!

I also spend some time trying to avoid using actionKeyContext - no good alternative so far - requires tooo many changes (like a lot). Will try add it to the BuildOptions maybe in the next experiment's round.